packages/web-config-server/src/models/Entity.js

```diff
@@ -4,228 +4,22 @@
  **/
 
 import { pascal } from 'case';
+import { EntityType as CommonEntityType, EntityModel as CommonEntityModel } from '@tupaia/database';
 
-import { TYPES } from '@tupaia/database';
-import { reduceToDictionary } from '@tupaia/utils';
 import {
   translateBoundsForFrontend,
   translatePointForFrontend,
   translateRegionForFrontend,
 } from '/utils/geoJson';
-import { BaseModel } from './BaseModel';
-import { AncestorDescendantRelation } from './AncestorDescendantRelation';
-import { Project } from './Project';
-
-const CASE = 'case';
-const CASE_CONTACT = 'case_contact';
-const COUNTRY = 'country';
-const DISASTER = 'disaster';
-const DISTRICT = 'district';
-const FACILITY = 'facility';
-const SCHOOL = 'school';
-const SUB_DISTRICT = 'sub_district';
-const CATCHMENT = 'catchment';
-const SUB_CATCHMENT = 'sub_catchment';
-const VILLAGE = 'village';
-const WORLD = 'world';
-const PROJECT = 'project';
-
-export const ENTITY_TYPES = {
-  CASE,
-  CASE_CONTACT,
-  COUNTRY,
-  DISASTER,
-  DISTRICT,
-  FACILITY,
-  SCHOOL,
-  SUB_DISTRICT,
-  CATCHMENT,
-  SUB_CATCHMENT,
-  VILLAGE,
-  WORLD,
-  PROJECT,
-};
-
-const ORG_UNIT_TYPE_LEVELS = {
-  [WORLD]: 1,
-  [COUNTRY]: 2,
-  [DISTRICT]: 3,
-  [SUB_DISTRICT]: 4,
-  [FACILITY]: 5,
-  [VILLAGE]: 6,
-};
-
-export class Entity extends BaseModel {
-  static databaseType = TYPES.ENTITY;
-
-  static fields = [
-    'id',
-    'code',
-    'type',
-    'parent_id',
-    'country_code',
-    'name',
-    'point',
-    'region',
-    'bounds',
-    'image_url',
-    'attributes',
-  ];
-
-  // a set of basic fields so that entities used for search etc. can be as light as possible
-  static minimalFields = ['id', 'code', 'type', 'parent_id', 'country_code', 'name'];
```

Comment:
This was previously used as a step to avoid out of memory errors - the idea being that we might have been caching too much information for each entity, and blowing our top that way. Since then, I've realised the out of memory was purely inside the postgres query, and likely had nothing to do with our cache. In any case, it was only ever used for ancestor fetch, not descendant. My strategy is to avoid adding the code complexity (and unpredictability of what data a model instance might hold) until we realise that we need it. I've done some performance testing with and without, and while there may be a bit more lag to fetch the geojson fields the first time, after the soft cache has kicked in it is the same speed with and without.

Reply:
I definitely agree with this strategy!